src/v/schema/registry.h

```diff
@@ -47,6 +47,16 @@ class registry {
     virtual ss::future<pandaproxy::schema_registry::schema_getter*>
     synced_getter() const = 0;
 
+    /// Synchronizes the underlying store if the time since the last sync
+    /// exceeds the specified threshold.
+    ///
+    /// \param max_age maximum duration allowed since last sync before forcing
+    ///                a new sync. If zero (default), always forces a sync.
+    /// \return time point of the last sync (either current time if sync was
+    ///         performed, or the previous sync time if no sync was needed).
+    virtual ss::future<ss::lowres_clock::time_point>
+    sync(ss::lowres_clock::duration max_age = {}) = 0;
```

Comment:
We have pretty effective caching here already right? Why do we need to complicate this? Can't we just always sync?

Reply:
For 1000 partitions do we want 1000 sync calls or 1?

Reply:
shouldn't our cache be per shard instead of per partition? Then we only sync as many cores as we have

Reply:
Ideally we should redo the schema id caching to capture the post-iceberg-conversion schema rather than pre. I ran some benchmarks with this new code vs value_schema_id_prefix which doesn't have the fancy cache and at least for protobuf there isn't a big perf difference.

I guess it does matter for JSON (for which we don't have a benchmark yet) because there the (by-)schema id cache does store the raw JSON (vs Descriptor in case of proto) which we need to parse/convert each time.